src/Icons/src/Iconify.php

```diff
@@ -40,13 +43,15 @@ public function __construct(
 
     public function metadataFor(string $prefix): array
     {
-        $response = $this->http->request('GET', sprintf('/collections?prefix=%s', $prefix));
-
-        return $response->toArray()[$prefix] ?? throw new \RuntimeException(sprintf('The icon prefix "%s" does not exist on iconify.design.', $prefix));
+        return $this->sets()[$prefix] ?? throw new \RuntimeException(sprintf('The icon prefix "%s" does not exist on iconify.design.', $prefix));
     }
 
     public function fetchIcon(string $prefix, string $name): Icon
     {
+        if (!isset($this->sets()[$prefix])) {
+            throw new IconNotFoundException(sprintf('The icon "%s:%s" does not exist on iconify.design.', $prefix, $name));
+        }
+
         $response = $this->http->request('GET', sprintf('/%s.json?icons=%s', $prefix, $name));
```

Comment:
If (i insist on _if_) this become the bottleneck, we have a card in hand: grouping icons per set   https://iconify.design/docs/api/icon-data.html#query